geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/RPopDUnitTest.java

```diff
@@ -0,0 +1,189 @@
+    List<String> listHashtags = makeListHashtags();
+    List<String> keys = makeListKeys(listHashtags);
+
+    List<String> elementList1 = makeElementList(keys.get(0), INITIAL_LIST_SIZE);
+    List<String> elementList2 = makeElementList(keys.get(1), INITIAL_LIST_SIZE);
+    List<String> elementList3 = makeElementList(keys.get(2), INITIAL_LIST_SIZE);
+
+    lpushPerformAndVerify(keys.get(0), elementList1);
+    lpushPerformAndVerify(keys.get(1), elementList2);
+    lpushPerformAndVerify(keys.get(2), elementList3);
+
+    Runnable task1 =
+        () -> rpopPerformAndVerify(keys.get(0), runningCount);
+    Runnable task2 =
+        () -> rpopPerformAndVerify(keys.get(1), runningCount);
+    Runnable task3 =
+        () -> rpopPerformAndVerify(keys.get(2), runningCount);
+
+    Future<Void> future1 = executor.runAsync(task1);
+    Future<Void> future2 = executor.runAsync(task2);
+    Future<Void> future3 = executor.runAsync(task3);
+
+    for (int i = 0; i < 50 && runningCount.get() > 0; i++) {
+      clusterStartUp.moveBucketForKey(listHashtags.get(i % listHashtags.size()));
+      Thread.sleep(500);
+    }
+
+    runningCount.set(0);
+
+    future1.get();
+    future2.get();
+    future3.get();
+  }
+
+  private List<String> makeListHashtags() {
+    List<String> listHashtags = new ArrayList<>();
+    listHashtags.add(clusterStartUp.getKeyOnServer("rpop", 1));
+    listHashtags.add(clusterStartUp.getKeyOnServer("rpop", 2));
+    listHashtags.add(clusterStartUp.getKeyOnServer("rpop", 3));
+    return listHashtags;
+  }
+
+  private List<String> makeListKeys(List<String> listHashtags) {
+    List<String> keys = new ArrayList<>();
+    keys.add(makeListKeyWithHashtag(1, listHashtags.get(0)));
+    keys.add(makeListKeyWithHashtag(2, listHashtags.get(1)));
+    keys.add(makeListKeyWithHashtag(3, listHashtags.get(2)));
+    return keys;
+  }
+
+  private void lpushPerformAndVerify(String key, List<String> elementList) {
+    jedis.lpush(key, elementList.toArray(new String[] {}));
+
+    Long listLength = jedis.llen(key);
+    assertThat(listLength).as("Initial list lengths not equal for key %s'", key)
+        .isEqualTo(elementList.size());
+  }
+
+  private void rpopPerformAndVerify(String key, AtomicLong runningCount) {
+    assertThat(jedis.llen(key)).isEqualTo(INITIAL_LIST_SIZE);
+
+    int elementCount = 0;
+    int elementListSize = INITIAL_LIST_SIZE - 1;
+    while (jedis.llen(key) > 0 && elementCount <= elementListSize && runningCount.get() > 0) {
+      String expected = "-" + key + "-" + elementCount + "-";
+      String element = expected;
+      try {
+        element = jedis.rpop(key);
+        if (element.equals(expected)) {
+          elementCount++;
+        } else {
+          int expectedCount = elementCount;
+          String[] chunks = element.split("-");
+          elementCount = Integer.parseInt(chunks[4]);
+          int actual = elementCount - expectedCount;
+          assertThat(actual).isLessThanOrEqualTo(1);
```

Comment:
This logic is different from the logic in the equivalent method in `LPopDUnitTest`, in that here we're saying that the actual element count of the popped element should not differ from the expected count by more than one, whereas in `LPopDUnitTest`, we have that the expected count is one less than the current element count AND that the actual element of the popped element should not differ from the expected count by more than one. This means that in the LPOP test, we actually tolerate differences of 1 - 2 in element count rather than 1 - 0 as we do in this test.

I think that to make it clearer exactly what we're expecting, the logic in both tests could be modified to be something like:
```
          // Allow the actual count to be exactly one greater than the expected count
          int allowedCount = elementCount + 1;
          String[] chunks = element.split("-");
          int actualCount = Integer.parseInt(chunks[4]);
          assertThat(actualCount).isEqualTo(allowedCount);
          // Set elementCount for the next element
          elementCount = actualCount + 1;
```
The final line here is important, as without it, `elementCount` will still be off by 1 on the next iteration of the loop, since we need to both increment elementCount each loop AND skip one element in the event that we got a duplicated pop due to client retries. Without this adjustment, if we ever got two duplicated pops in the same test, the test would fail, despite this logic being intended to make the test tolerant of client retries.

Alternately, this logic could be refactored to avoid the use of an if statement and using `satisfiesAnyOf()` instead:
```
      try {
        element = jedis.rpop(key);
        String[] chunks = element.split("-");
        int actualCount = Integer.parseInt(chunks[4]);
        int expectedCount = elementCount;
        assertThat(actualCount)
            .as("Actual element count did not match expected element count for element " + element)
            .satisfiesAnyOf(
                count -> assertThat(count).isEqualTo(expectedCount),
                count -> assertThat(count).isEqualTo(expectedCount + 1));
        elementCount = actualCount + 1;
      } catch (Exception ex) {
```